### PR TITLE
BOM-1197

### PR DESCRIPTION
**Pull request by Ayub-Khan**
-unpinned xss-utils version
-pinned edx-rest-api-client version
-updated transifex constraints
-removed python2.7 support
-added python3.6 support
-quality fixes for python3.5

### CODE REVIEW

**Review by UsamaSadiq — APPROVED** [review]

File: .travis.yml [hunk 1]
```diff
@@ -36,18 +36,18 @@ script:
...
+    - python: 3.5
       env:
         TESTNAME=quality-and-js
-        TARGETS="PYTHON_ENV=py27 requirements.js check_translations_up_to_date validate_translations clean_static static quality validate_js"
-    - python: 2.7
-      env: TESTNAME=test-python TARGETS="PYTHON_ENV=py27 requirements.js clean_static static validate_python"
+        TARGETS="PYTHON_ENV=py35 requirements.js check_translations_up_to_date validate_translations clean_static static quality validate_js"
     - python: 3.5
+      env: TESTNAME=test-python TARGETS="PYTHON_ENV=py35 requirements.js clean_static static validate_python"
+    - python: 3.6
```
Comment (jmbowman) [inline .travis.yml:46]: Why 3.6 instead of 3.5?  I think ecommerce is still using 3.5 for both development and production.

File: ecommerce/extensions/api/v2/tests/views/test_enterprise.py [hunk 2]
```diff
@@ -511,7 +511,6 @@ def assert_coupon_codes_response(self, response, coupon_id, max_uses, results_co
         """
         coupon = Product.objects.get(id=coupon_id)
         all_coupon_codes = coupon.attr.coupon_vouchers.vouchers.values_list('code', flat=True)
-        all_coupon_codes = [code for code in all_coupon_codes]
```
Comment (awais786) [inline ecommerce/extensions/api/v2/tests/views/test_enterprise.py:514]: why did you remove this line ?

File: ecommerce/extensions/voucher/utils.py [hunk 3]
```diff
@@ -872,27 +870,25 @@ def get_voucher_discount_info(benefit, price):
...
+        if discount_percentage > 100:
+            discount_percentage = 100.00
+            discount_value = price
         else:
-            discount_percentage = get_discount_percentage(discount_value=benefit_value, product_price=price)
-            if discount_percentage > 100:
-                discount_percentage = 100.00
-                discount_value = price
-            else:
-                discount_percentage = discount_percentage
```
Comment (awais786) [inline ecommerce/extensions/voucher/utils.py:883]: You missed this line ?

File: ecommerce/extensions/api/v2/tests/views/test_enterprise.py (hunk 2 above)
Comment (Ayub-Khan) [reply ecommerce/extensions/api/v2/tests/views/test_enterprise.py:514]: It's not doing anything.

File: ecommerce/extensions/voucher/utils.py (hunk 3 above)
Comment (Ayub-Khan) [reply ecommerce/extensions/voucher/utils.py:883]: no. It's not doing anything.

**Review by awais786 — APPROVED** [review]

File: requirements/docs.txt [hunk 4]
```diff
@@ -5,19 +5,22 @@
 #    make upgrade
 #
 alabaster==0.7.12         # via sphinx
-babel==2.7.0              # via sphinx
+babel==2.8.0              # via sphinx
 certifi==2019.11.28       # via requests
 chardet==3.0.4            # via requests
-docutils==0.15.2          # via sphinx
+docutils==0.16            # via sphinx
+edx-rest-api-client==1.9.2
```
Comment (jmbowman) [inline requirements/docs.txt]: This was pulled in because it was added to `pins.txt`; could you update all the requirements files with `-r pins.txt` to use `-c pins.txt` instead?  This imposes the version constraints if the package is present, but won't install the constrained packages if they aren't actually needed.

**Review by jmbowman — COMMENTED** [review]
Looks good to me other than the 2 noted points.  @edx/revenue-squad This does seem like a lot of dependencies to upgrade at once, even if most of them are fairly small upgrades.  Are you okay with merging this all at once, or would you feel more comfortable splitting the upgrades up into multiple PRs across several deployments?

File: .travis.yml (hunk 1 above)
Comment (Ayub-Khan) [reply .travis.yml:46]: oh I just moved it to first and added py3.6. before that py27 was the first.

**Comment by awais786** [comment]
@jmbowman any update on this ?

File: .travis.yml [hunk 5]
```diff
@@ -36,18 +37,18 @@ script:
...
+    - python: 3.5
       env:
         TESTNAME=quality-and-js
-        TARGETS="PYTHON_ENV=py27 requirements.js check_translations_up_to_date validate_translations clean_static static quality validate_js"
-    - python: 2.7
-      env: TESTNAME=test-python TARGETS="PYTHON_ENV=py27 requirements.js clean_static static validate_python"
+        TARGETS="requirements.js check_translations_up_to_date validate_translations clean_static static quality validate_js"
     - python: 3.5
+      env: TESTNAME=test-python TARGETS="requirements.js clean_static static validate_python"
+    - python: 3.6
```
Comment (robrap) [inline .travis.yml:46]: @jmbowman: Should we be targeting python 3.5 and 3.8 only for all repos?

Comment (jmbowman) [reply .travis.yml:46]: TBD.  Most are currently using 3.5 in production since that comes with Ubuntu 16.04, but it's looking like some services may move to 18.04 (which comes with Python 3.6) and others may skip directly to 20.04 (which will probably come with Python 3.8).

**Comment by jmbowman** [comment]
@emma-green Any thoughts on my question above regarding whether it's ok to merge this at one go or it should be split up?  This is the current blocker on most Django 2.2 upgrade work in ecommerce.

**Review by jmbowman — APPROVED** [review]
I contacted Emma, she said it's ok to merge this as long as the release gets monitored for problems.  Go ahead and fix the merge conflicts, and let's target merging tomorrow morning Boston time.
